ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -5,102 +5,156 @@
+        return struct.pack(">BHB", MessageID.MOVE.message_id, hw_id, self.direction)
 
 
 @dataclass
 class SyncPinMessage(Message):
     """Message for setting sync pin high or low."""
 
-    state: Literal["HIGH", "LOW"]
-
-    @classmethod
-    def to_message(cls, string: str) -> Optional[SyncPinMessage]:
-        """Parse string into SyncPinMessage or None."""
-        if string.startswith("SYNC_PIN"):
-            split_string = string.split(" ")
-            if len(split_string) == 2:
-                sync_pin_state = split_string[1]
-                if sync_pin_state == "HIGH":
-                    return cls(state="HIGH")
-                elif sync_pin_state == "LOW":
-                    return cls(state="LOW")
-
-        return None
-
-
-@dataclass
-class MoveMessage(Message):
-    """Message to be sent between hardware simulators and OT3StateManager"""
-
-    axis: OT3Axis
-    direction: Direction
+    state: SyncPinState
 
-    @classmethod
-    def to_message(cls, string: str) -> Optional[MoveMessage]:
-        """Parse string to Message or None."""
-        split_string = string.split(" ")
+    def __eq__(self, other: Any) -> bool:
+        """Confirm that the 2 objects are equal"""
+        return isinstance(other, SyncPinMessage) and other.state == self.state
 
-        if len(split_string) == 2:
-            try:
-                axis = OT3Axis[split_string[1]]
-            except KeyError:
-                axis = None
+    def __hash__(self) -> int:
+        """Return hash for instance."""
+        return hash(str(self))
 
-            try:
-                direction = Direction.from_symbol(split_string[0])
-            except ValueError:
-                direction = None
+    @staticmethod
+    def build(message_content: bytes) -> SyncPinMessage:
+        """Convert message_content into a SyncPinMessage object."""
+        # Have to use from_bytes method because struct package won't accept a 3 byte val
+        val = int.from_bytes(message_content, "big")
+        state = SyncPinState.from_int(val)
+        return SyncPinMessage(state=state)
 
-            if axis is not None and direction is not None:
-                return cls(axis=axis, direction=direction)
+    def to_bytes(self) -> bytes:
+        """Convert SyncPinMessage object into a sequence of hexadecimal bytes."""
+        # 3rd arg is 0 because middle two bytes are not used for SyncPinMessage
+        return struct.pack(">BHB", MessageID.SYNC_PIN.message_id, 0, self.state)
 
-        return None
 
+class MessageID(Enum):
+    """Enum class defining the relationship between the message_id byte and the corresponding Message object."""
 
-def parse_message(string: str) -> Union[SyncPinMessage, MoveMessage]:
-    """Parse string into a Message Type"""
-    sync_pin_message = SyncPinMessage.to_message(string)
-    move_message = MoveMessage.to_message(string)
+    def __new__(cls, message_id: int, builder_func: Callable) -> MessageID:
```

Comment:
we probably want a mypy protocol or some other richer type to describe callable; that's likely the source of the type confusion above.

Reply:
Instead of doing a mypy protocol, I just passed the whole Message object, which requires a `build_message` method. And in the __init__ set the builder_func to the `build_message` method.